server/src/main/java/org/apache/iotdb/db/engine/TsFileMetricManager.java

```diff
@@ -50,30 +71,154 @@ public class TsFileMetricManager {
   private final AtomicInteger innerSeqCompactionTempFileNum = new AtomicInteger(0);
   private final AtomicInteger innerUnseqCompactionTempFileNum = new AtomicInteger(0);
   private final AtomicInteger crossCompactionTempFileNum = new AtomicInteger(0);
+  private AbstractMetricService metricService;
+  private AtomicBoolean hasRemainData = new AtomicBoolean(false);
 
   private TsFileMetricManager() {}
 
   public static TsFileMetricManager getInstance() {
     return INSTANCE;
   }
 
-  public void addFile(long size, boolean seq) {
+  public void setMetricService(AbstractMetricService metricService) {
+    this.metricService = metricService;
+  }
+
+  public void addFile(long size, boolean seq, String name) {
+    updateGlobalCountAndSize(size, 1, seq);
+    try {
+      TsFileNameGenerator.TsFileName tsFileName = TsFileNameGenerator.getTsFileName(name);
+      int level = tsFileName.getInnerCompactionCnt();
+      updateLevelCountAndSize(size, 1, seq, level);
+    } catch (IOException e) {
+      log.error("Unexpected error occurred when getting tsfile name", e);
+    }
+  }
+
+  private void updateGlobalCountAndSize(long sizeDelta, int countDelta, boolean seq) {
     if (seq) {
-      seqFileSize.getAndAdd(size);
-      seqFileNum.incrementAndGet();
+      seqFileSize.getAndAdd(sizeDelta);
+      seqFileNum.getAndAdd(countDelta);
     } else {
-      unseqFileSize.getAndAdd(size);
-      unseqFileNum.incrementAndGet();
+      unseqFileSize.getAndAdd(sizeDelta);
+      unseqFileNum.getAndAdd(countDelta);
     }
   }
 
-  public void deleteFile(long size, boolean seq, int num) {
+  private void updateLevelCountAndSize(long sizeDelta, int countDelta, boolean seq, int level) {
+    int count = 0;
+    long totalSize = 0;
     if (seq) {
-      seqFileSize.getAndAdd(-size);
-      seqFileNum.getAndAdd(-num);
+      count =
+          seqLevelTsFileCountMap.compute(level, (k, v) -> v == null ? countDelta : v + countDelta);
+      totalSize =
+          seqLevelTsFileSizeMap.compute(level, (k, v) -> v == null ? sizeDelta : v + sizeDelta);
+    } else {
+      count =
+          unseqLevelTsFileCountMap.compute(
+              level, (k, v) -> v == null ? countDelta : v + countDelta);
+      totalSize =
+          unseqLevelTsFileSizeMap.compute(level, (k, v) -> v == null ? sizeDelta : v + sizeDelta);
+    }
+    updateLevelFileInfoInMetricService(totalSize, count, seq, level);
+  }
+
+  private void updateLevelFileInfoInMetricService(
+      long totalSize, int count, boolean seq, int level) {
+    if (metricService != null) {
+      metricService
+          .getOrCreateGauge(
+              FILE_LEVEL_COUNT,
+              MetricLevel.CORE,
+              Tag.TYPE.toString(),
+              seq ? SEQUENCE : UNSEQUENCE,
+              LEVEL,
+              String.valueOf(level))
+          .set(count);
```

Comment:
Consider avoiding the acquire of gauge from metric service. Instead, we can create a metric set to manage all related metrics, including creation, update and remove, more details: https://github.com/apache/iotdb/pull/9901